File: kyw0716/level0/유한소수 판별하기.js
const getYaksoo = (n) => {
  const yaksoo = [];

  for (let i = n; i >= 1; i--) {
    if (n % i === 0) yaksoo.push(i);
  }

  return yaksoo;
};

const getIsGiYak = (n) => {
  while (n % 5 === 0) {
    n /= 5;
  }

  while (n % 2 === 0) {
    n /= 2;
  }

  return n === 1;
};

function solution(a, b) {
  const aYaksoo = getYaksoo(a);
  const bYaksoo = getYaksoo(b);
  let maxCommonYaksoo;

  for (let i = 0; i < aYaksoo.length; i++) {
    if (bYaksoo.includes(aYaksoo[i])) {
      maxCommonYaksoo = aYaksoo[i];
      break;
    }
  }

  return getIsGiYak(b / maxCommonYaksoo) ? 1 : 2;
}

// https://school.programmers.co.kr/learn/courses/30/lessons/120878

File: sk000801/입문/유한소수 판별하기.java
//일단 최대공약수 구하기
//최대공약수를 분모와 나누기
//2,5로 나눠서 1이 되면 유한, 안되면 무한 (애초에 1이면 정수이므로 가눙)
class Solution {
    public int solution(int a, int b) {
        int max = 0; //최대 공약수
		for(int i=1; i<=a && i<=b; i++) {
			if(a%i==0 && b%i==0) {
				max = i;
			}
		}
        b /= max;
        while(b%2==0) {b/=2;}
        while(b%5==0) {b/=5;}
        return b == 1 ?  1 :  2;
    }
}